frontend/src/schemas/SchemaList.tsx

```diff
@@ -19,7 +24,7 @@ export default function SchemaList({ schemaKind }: SchemaDisplayProps) {
     <>
       {isSchemasLoading && <Loading />}
       <List>
-        {schemas.map((schema, index) => (
+        {schemaArray.map((schema, index) => (
```

Comment:
Sorry my previous comment was unclear, that's on me. I meant to `useMemo` the `.map` of the array as that's where the expensive DOM changes lie.
```
  const schemaList = useMemo(
    () =>
      schemas.map((schema, index) => (
        <ListItem
          key={schema.id}
          divider={index + 1 !== schemas.length}
          disableGutters
          secondaryAction={
            <>
              <Button disabled>Edit</Button>
              <Button disabled>{schema.active ? 'Set as inactive' : 'Mark as active'}</Button>
            </>
          }
        >
          <ListItemText primary={schema.name} sx={{ maxWidth: 'fit-content', wordBreak: 'break-all', pr: 20 }} />
        </ListItem>
      )),
    [schemas],
  )
```